lib/rake/application.rb

```diff
@@ -269,18 +272,11 @@ def has_chain?(exception) # :nodoc:
     end
     private :has_chain?
 
-    # True if one of the files in RAKEFILES is in the current directory.
-    # If a match is found, it is copied into @rakefile.
-    def have_rakefile # :nodoc:
-      @rakefiles.each do |fn|
-        if File.exist?(fn)
-          others = FileList.glob(fn, File::FNM_CASEFOLD)
-          return others.size == 1 ? others.first : fn
-        elsif fn == ""
-          return fn
-        end
+    # Returns first filename from @rakefiles that exists in the specified dir.
+    def have_rakefile(dir = Dir.pwd) # :nodoc:
+      Dir.chdir(dir) do
+        Dir.glob(@rakefiles.map { |name| escape_glob(name) }).first || @rakefiles.find(&:empty?)
```

Comment:
When support for Ruby < 2.5 is dropped, this `Dir.chdir` can be replaced with [`Dir.glob`'s `base` kwarg](https://docs.ruby-lang.org/en/2.5.0/Dir.html#method-c-glob).